extensions/resource-deployment/src/ui/resourceTypePickerDialog.ts

```diff
@@ -297,6 +299,16 @@ export class ResourceTypePickerDialog extends DialogBase {
 		return createFlexContainer(this._view, [checkbox, text]);
 	}
 
+	private getAgreementDisplayText(agreementInfo: AgreementInfo): string {
+		// the agreement template will have {index} as placeholder for hyperlinks
+		// this method will get the display text after replacing the placeholders
+		let text = agreementInfo.template;
```

Comment:
This is fine - but in general for this kind of thing you could also use the aria-labeledby attribute to reduce the amount of duplication of logic